### PR TITLE
Feature/plugin tad

### PR DESCRIPTION
**Pull request by kylebaron**
#702 

As a user, I want to be able to calculate the time after dose for specific dosing compartments. 

This functionality is exposed by invoking the `tad` plugin (`$PLUGIN tad`)

This implements a `tadose` class whose objects can track specific compartment numbers.  To track time after dose, create an object and register the compartment number to track.  When the `.tad()` method is called, it will return the time after dose.  Time after dose is -1 until the first dose is administered.

### CODE REVIEW

**Comment by kylebaron** [comment]
@yizhang-yiz does technical review, @timwaterhouse does functional review.  Please respond here about when you can get this review done.

**Comment by yizhang-yiz** [comment]
I can have it done by end of day.

**Review by timwaterhouse — APPROVED** [review]
Functionality looks good.

File: inst/base/mrgsolve-tad.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  double tad(databox& self);
+  void reset();
+};
+/**
+ * Constructor.
+ * 
+ * @param cmt_ compartment number
+ * 
+ */
+tadose::tadose(int cmt_) {
```
Comment (yizhang-yiz) [inline inst/base/mrgsolve-tad.h:65]: In general it's not a good idea to leave things uninitialized. Above all it simply makes it hard to debug. So for `cmt` we can give it a large number, such as `std::numeric_limits<int>`. For `had_dose` we need to set it to `false`.

File: inst/base/mrgsolve-tad.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ * 
+ * @param self the model `databox` object
+ * 
+ * @return the time after dose or `-1` if no dose has been given
+ * 
+ */
+double tadose::tad(databox& self) {
+  if(self.newind <= 1) this->reset();
+  if(self.evid==0) {
+    return had_dose ? self.time - told : -1.0;
```
Comment (yizhang-yiz) [inline inst/base/mrgsolve-tad.h:90]: Is there a meaning of `time=-1.0`? If so we need use something else. 

This line does not necessarily require any action.

**Review by yizhang-yiz — APPROVED** [review]
Everything's good all around. Only a minor suggestion.

File: inst/base/mrgsolve-tad.h (hunk 1 above)
Comment (kylebaron) [reply inst/base/mrgsolve-tad.h:66]: I should just include a comment like this in a PR review template so you don't have to type it out every single review :). 

1. Providing default values for both `cmt` and `had_dose`
1. Including default initialization in the test model

File: inst/base/mrgsolve-tad.h (hunk 2 above)
Comment (kylebaron) [reply inst/base/mrgsolve-tad.h:83]: I use -1 in other time after dose functionality; so that has come to mean there hasn't been a dose yet.

**Review by yizhang-yiz — APPROVED** [review]

**Pull request merged**